src/main/java/uk/ac/ebi/subs/validator/coordinator/SampleValidationMessageEnvelopeExpander.java

```diff
@@ -37,4 +51,27 @@ void expandEnvelope(SampleValidationMessageEnvelope validationMessageEnvelope) {
             }
         }
     }
+
+    private Sample findSampleByAccession(String accessionID) {
+        Sample sample = sampleRepository.findByAccession(accessionID);
+
+        if (sample == null) {
+            try {
+                final ResponseEntity<String> response =
+                        restTemplate.getForEntity(sampleStorageURI + "samples/" + accessionID, String.class);
```

Comment:
I would suggest creating a new service and moving this over there.

Reply:
I also thought about it, but I think I would do it when we are going to do it for the other submittable. I did not want to introduce a new service just for the sample search. What do you think?

Reply:
OK. I moved it to another service.